### PR TITLE
Transaction: add back 2 deprecated constructors

### PR DESCRIPTION
**Pull request by msgilligan**
Move most of `Transaction.read` into a private constructor. Use this constructor to add back 2 removed constructors as deprecated constructors.

### CODE REVIEW

File: core/src/main/java/org/bitcoinj/core/Transaction.java [hunk 1]
```diff
@@ -329,6 +332,18 @@ public Transaction(NetworkParameters params) {
         this();
     }
 
+    /** @deprecated use {@link Transaction#readOutputs(ByteBuffer)} */
+    @Deprecated
+    public Transaction(NetworkParameters parameters, byte[] payload) {
+        this(ByteBuffer.wrap(payload), ProtocolVersion.CURRENT.intValue());
```
[the remark this answers is not in the document]
Comment (msgilligan) [reply core/src/main/java/org/bitcoinj/core/Transaction.java:338]: Maybe we should reduce the scope of this PR to just the refactoring that moves most of `read()` into a private constructor?

[the remark this answers is not in the document]
Comment (schildbach) [reply core/src/main/java/org/bitcoinj/core/Transaction.java:338]: Would this be enough to cover the need of LDK? Then sure, the less changes the better.

[the remark this answers is not in the document]
Comment (msgilligan) [reply core/src/main/java/org/bitcoinj/core/Transaction.java:338]: Moving most of `read()` into a private constructor does not change the public API in any way, so it won't help LDK. It only makes any subsequent changes to constructors easier.

**Review by schildbach — COMMENTED** [review]
I just noticed that for some reason, my comments belonging to this (second) review were still pending, so I assume not visible.

**Comment by schildbach** [comment]
I tend to _not_ merge this. I haven't heard any complaining about moved or changed API, so I assume there is not much benefit merging this. On the other hand, the potential semantics change (see review comments) bears some risk I'd like to avoid.

**Comment by schildbach** [comment]
Closing for now.

**Pull request closed without merging**